src/BotPlutusInterface/Files.hs

```diff
@@ -176,15 +177,21 @@ readPrivateKeys ::
   Eff effs (Either Text (Map PubKeyHash DummyPrivKey))
 readPrivateKeys pabConf = do
   files <- listDirectory @w $ Text.unpack pabConf.pcSigningKeyFileDir
-  let vKeyFiles =
-        map (\filename -> Text.unpack pabConf.pcSigningKeyFileDir ++ "/" ++ filename) $
-          filter ("vkey" `isExtensionOf`) files
-  let sKeyFiles =
-        map (\filename -> Text.unpack pabConf.pcSigningKeyFileDir ++ "/" ++ filename) $
-          filter ("skey" `isExtensionOf`) files
-  privKeys <- mapM (readVerificationKey @w) vKeyFiles
-  privKeys' <- mapM (readSigningKey @w) sKeyFiles
-  pure $ toPrivKeyMap <$> sequence (privKeys <> privKeys')
+
+  privKeys <-
+    catMaybes
+      <$> mapM
+        ( \filename ->
+            let fullPath = Text.unpack pabConf.pcSigningKeyFileDir </> filename
+             in case filename of
+                  _
+                    | "vkey" `isExtensionOf` filename -> Just <$> readVerificationKey @w fullPath
+                    | "skey" `isExtensionOf` filename -> Just <$> readSigningKey @w fullPath
+                    | otherwise -> pure Nothing
```

Comment:
Theres a function to extract the extension of a file, maybe case over that instead of a guards only case?